packages/hebao_v1/contracts/test/DummyExchange.sol

```diff
@@ -0,0 +1,182 @@
+    event Deposit(address accountAddr, address token, uint amount);
+    event Withdrawal(address accountAddr, address token, uint amount);
+
+    struct Account {
+        uint24 accountId;
+        uint pubKeyX;
+        uint pubKeyY;
+    }
+
+    function isInWithdrawalMode()
+        external
+        view
+        override
+        returns (bool) {
+        return false;
+    }
+
+    function isShutdown()
+        external
+        view
+        override
+        returns (bool) {
+        return false;
+    }
+
+    function isInMaintenance()
+        external
+        view
+        override
+        returns (bool) {
+        return false;
+    }
+
+    function getAccount(
+        address owner
+        )
+        external
+        view
+        override
+        returns (
+            uint24 accountID,
+            uint   pubKeyX,
+            uint   pubKeyY
+        ) {
+        Account memory account = accountMap[owner];
+        accountID = account.accountId;
+        pubKeyX = account.pubKeyX;
+        pubKeyY = account.pubKeyY;
+    }
+
+    function createOrUpdateAccount(
+        uint  pubKeyX,
+        uint  pubKeyY,
+        bytes calldata permission
+        )
+        external
+        payable
+        override
+        returns (
+            uint24 accountID,
+            bool   isAccountNew,
+            bool   isAccountUpdated
+        ) {
+        if (accountMap[msg.sender].accountId == 0) {
+            isAccountNew = true;
+            isAccountUpdated = false;
+            accountID = lastAccountId + 1;
+            accountMap[msg.sender] = Account(accountID, 0, 0);
+            emit AccountCreated(accountID, msg.sender);
+        } else {
+            accountID = accountMap[msg.sender].accountId;
+            isAccountNew = true;
+            isAccountUpdated = false;
+        }
+
+    }
+
+    function updateAccountAndDeposit(
+        uint    pubKeyX,
+        uint    pubKeyY,
+        address tokenAddress,
+        uint96  amount,
+        bytes   calldata permission
+        )
+        external
+        payable
+        override
+        returns (
+            uint24 accountID,
+            bool   isAccountNew,
+            bool   isAccountUpdated
+        ) {
+        if (accountMap[msg.sender].accountId == 0) {
+            isAccountNew = true;
+            isAccountUpdated = false;
+            accountID = lastAccountId + 1;
+            accountMap[msg.sender] = Account(accountID, 0, 0);
+        } else {
+            accountID = accountMap[msg.sender].accountId;
+            isAccountNew = true;
+            isAccountUpdated = false;
+        }
+
+    }
+
+    function deposit(
+        address tokenAddress,
+        uint96  amount
+        )
+        external
+        payable
+        override
+    {
+
+    }
+
+    function withdraw(
+        address tokenAddress,
+        uint96  amount
+        )
+        external
+        payable
+        override
+    {
+
```

Comment:
Looks we should emit Withdraw event.

Reply:
All the events I checked in test/testLoopringModule.ts were emitted in LoopringModule.sol.
So I think I will just remove all the events in DummyExchange.sol